network/endpoint_linux.go

```diff
@@ -54,204 +55,203 @@ func (nw *network) newEndpointImpl(
 	plc platform.ExecClient,
 	netioCli netio.NetIOInterface,
 	epClient EndpointClient,
-	epInfo *EndpointInfo,
+	epInfo []*EndpointInfo,
 ) (*endpoint, error) {
-	var containerIf *net.Interface
-	var ns *Namespace
-	var ep *endpoint
-	var err error
-	var hostIfName string
-	var contIfName string
-	var localIP string
-	var vlanid int = 0
-
-	if nw.Endpoints[epInfo.Id] != nil {
-		log.Printf("[net] Endpoint alreday exists.")
-		err = errEndpointExists
-		return nil, err
+	var (
+		containerIf *net.Interface
+		err         error
+		hostIfName  string
+		contIfName  string
+		localIP     string
+		vlanid      = 0
+		ep          = &endpoint{}
+	)
+	if nw.extIf != nil {
+		ep.Gateways = []net.IP{nw.extIf.IPv4Gateway}
 	}
 
-	if epInfo.Data != nil {
-		if _, ok := epInfo.Data[VlanIDKey]; ok {
-			vlanid = epInfo.Data[VlanIDKey].(int)
-		}
-
-		if _, ok := epInfo.Data[LocalIPKey]; ok {
-			localIP = epInfo.Data[LocalIPKey].(string)
-		}
-	}
+	for _, epInfo := range epInfo {
+		epClient := epClient
```

Comment:
can we give different name..otherwise it will be confusing

Reply:
ok will update